### ISSUE TITLE
Exponential backoff

### ISSUE THREAD

**Issue by moubry** [rest-client/rest-client#428]
Whenever you make HTTP requests, there are plenty of completely valid reasons for the request to not complete:

From [this book](https://www.safaribooksonline.com/library/view/http-the-definitive/1565925092/ch04s07.html):

> Connections can close at any time, even in non-error conditions. HTTP applications have to be ready to properly handle unexpected closes.

In this situation—and others like it (like a timeout), I'd expect most developers would want to immediately retry the request.

What do y'all think about adding an [exponential backoff](https://en.wikipedia.org/wiki/Exponential_backoff) feature to retry failed requests? Seems like something you'd always want turned on by default too, right?

Somewhat related issue: https://github.com/rest-client/rest-client/issues/108

**Comment by ab** [comment]
Have you tried something like https://github.com/ooyala/retries ?

**Comment by moubry** [comment]
Yes. Which I'm fine with. I just wanted to check and see if it would be appropriate to do this by default in RestClient itself.

**Comment by ab** [comment]
Yeah it's definitely plausible. I think Python's urllib3 does do retries. Main wrinkle I can think of in implementing it would be how / whether to capture intermediate failed responses so that they're visible to the caller.
